proof/refine/InvariantsPre_H.thy

```diff
@@ -0,0 +1,129 @@
+(*
+ * Copyright 2023, Proofcraft Pty Ltd
+ * Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ *)
+
+(* Preliminary definitions required to define arch-specific invariants and properties.
+   These are primarily related to obj_at' and storability.
+*)
+
+theory InvariantsPre_H
+imports
+  LevityCatch
+  "AInvs.ArchDetSchedSchedule_AI"
+  "Lib.Heap_List"
+  Move_R
+begin
+
+(* global data and code of the kernel, not covered by any cap *)
+axiomatization
+  kernel_data_refs :: "word64 set"
+
+
+section \<open>Locale Setup for kernel_state Field Update Identities\<close>
+
+text \<open>
+  These locales identify which updates are the identity function on specific field access
+  predicates. For instance, when a function doesn't change the @{term ksPSpace} component of the
+  state, that function is the identity on all predicates that only depend on @{term ksPSpace}.\<close>
+
+text \<open>
+  Since locales can dynamically gain conclusions, we want to avoid situations where someone adds
+  a conclusion somewhere in the locale graph and is surprised that it does not appear for
+  extensions of that locale. For arch-split there is an extra complication, where the conclusions
+  might use arch-specific facts, thus needing to be in Arch.
+
+  This means that for every one the update locales, we have to add an additional Arch locale.
```

Comment:
```suggestion
  This means that for every one of the update locales, we have to add an additional Arch locale.
```

Reply:
Thanks, fixed, will include in next round.